[expo-image] NoSuchMethodException setLayoutDirectionOverride

### Minimal reproducible example

https://github.com/Bardiamist/diff/tree/expo-image-set-layout-direction-override

### What platform(s) does this occur on?

Android

### Where did you reproduce the issue?

in a standalone app

### Summary

I updated bare project to react native 0.75
expo-image: 1.12.14

Result: crash in release build

`nm4.setLayoutDirectionOverride [int]`

```
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:562)
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
Caused by: java.lang.reflect.InvocationTargetException
  at java.lang.reflect.Method.invoke(Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971) 
Caused by: java.lang.NoSuchMethodException: nm4.setLayoutDirectionOverride [int]
  at java.lang.Class.getMethod(Class.java:2937)
  at java.lang.Class.getMethod(Class.java:2449)
  at expo.modules.image.ExpoImageView.onDraw(Unknown Source:62)
  at android.view.View.draw(View.java:23889)
  at expo.modules.image.ExpoImageView.draw(Unknown Source:73)
  at android.view.View.updateDisplayListIfDirty(View.java:22756)
    ...
 ```
 
 Looks similar: https://github.com/expo/expo/issues/31112
 

### Environment

```
  expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.6.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 22.6.0 - /opt/homebrew/bin/node
      Yarn: 4.4.0 - /opt/homebrew/bin/yarn
      npm: 10.8.2 - /opt/homebrew/bin/npm
      Watchman: 2024.08.19.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.12.1 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.5, iOS 17.5, macOS 14.5, tvOS 17.5, visionOS 1.2, watchOS 10.5
    IDEs:
      Android Studio: 2024.1 AI-241.18034.62.2411.12169540
      Xcode: 15.4/15F31d - /usr/bin/xcodebuild
    npmPackages:
      expo: 51.0.29 => 51.0.29 
      react: 18.3.1 => 18.3.1 
      react-native: 0.75.2 => 0.75.2 
    npmGlobalPackages:
      expo-cli: 5.4.3
    Expo Workflow: bare
```

### Expo Doctor Diagnostics

```
✔ Check Expo config for common issues
✔ Check package.json for common issues
✔ Check for app config fields that may not be synced in a non-CNG project
✔ Check dependencies for packages that should not be installed directly
✔ Check for common project setup issues
✖ Check for issues with metro config
✔ Check npm/ yarn versions
✔ Check native tooling versions
✖ Check Expo config (app.json/ app.config.js) schema
✔ Check for legacy global CLI installed locally
✔ Check that native modules do not use incompatible support packages
✖ Check that packages match versions required by installed Expo SDK
✔ Check that native modules use compatible support package versions for installed Expo SDK

Detailed check results:

Error: Problem validating fields in /Users/bardiamist/Documents/proj/proj-mobile/app.json. Learn more: https://docs.expo.dev/workflow/configuration/
 • should NOT have additional property 'displayName'.

It looks like that you are using a custom metro.config.js that does not extend @expo/metro-config. This can lead to unexpected and hard to debug issues. Learn more: https://docs.expo.dev/guides/customizing-metro/
Advice: Update your custom metro.config.js to extend @expo/metro-config.

(node:18865) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
The following packages should be updated for best compatibility with the installed expo version:
  @react-native-async-storage/async-storage@1.24.0 - expected version: 1.23.1
  @react-native-community/netinfo@11.3.2 - expected version: 11.3.1
  @sentry/react-native@5.30.0 - expected version: ~5.22.0
  @shopify/flash-list@1.7.1 - expected version: 1.6.4
  lottie-react-native@6.7.2 - expected version: 6.7.0
  react@18.3.1 - expected version: 18.2.0
  react-native@0.75.2 - expected version: 0.74.5
  react-native-gesture-handler@2.18.1 - expected version: ~2.16.1
  react-native-pager-view@6.3.4 - expected version: 6.3.0
  react-native-reanimated@3.15.0 - expected version: ~3.10.1
  react-native-safe-area-context@4.10.9 - expected version: 4.10.5
  react-native-screens@3.34.0 - expected version: 3.31.1
  react-native-svg@15.5.0 - expected version: 15.2.0
  react-native-webview@13.11.1 - expected version: 13.8.6
  @types/react@18.3.4 - expected version: ~18.2.79
  typescript@5.5.4 - expected version: ~5.3.3
Your project may not work correctly until you install the expected versions of the packages.
Found outdated dependencies
```

@Bardiamist Running into this too on RN `0.75.2`, `0.74.5` works fine. 

Using expo 51.0.30 with old arch and proguard. This only happens in release builds on android.

Logcat
```bash
2024-08-22 20:53:21.679 10663-10663 SearchResultIcon        pid-10663                            D  Launching Omnid icon with component ComponentInfo{com.cupoc.omnid/com.cupoc.omnid.MainActivity}
2024-08-22 20:53:21.697  1724-27338 ActivityTaskManager     pid-1724                             I  START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.cupoc.omnid/.MainActivity bnds=[53,300][248,525]} with LAUNCH_SINGLE_TASK from uid 10227 (BAL_ALLOW_ALLOWLISTED_COMPONENT) result code=0
2024-08-22 20:53:21.698  2540-2579  WindowManagerShell      pid-2540                             V  Transition requested (#15063): android.os.BinderProxy@93dc852 TransitionRequestInfo { type = OPEN, triggerTask = TaskInfo{userId=0 taskId=10691 displayId=0 isRunning=true baseIntent=Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.cupoc.omnid/.MainActivity } baseActivity=ComponentInfo{com.cupoc.omnid/com.cupoc.omnid.MainActivity} topActivity=ComponentInfo{com.cupoc.omnid/com.cupoc.omnid.MainActivity} origActivity=null realActivity=ComponentInfo{com.cupoc.omnid/com.cupoc.omnid.MainActivity} numActivities=1 lastActiveTime=2860245345 supportsMultiWindow=true resizeMode=1 isResizeable=true minWidth=-1 minHeight=-1 defaultMinSize=220 token=WCT{android.window.IWindowContainerToken$Stub$Proxy@3c37723} topActivityType=1 pictureInPictureParams=null shouldDockBigOverlays=false launchIntoPipHostTaskId=-1 lastParentTaskIdBeforePip=-1 displayCutoutSafeInsets=Rect(0, 132 - 0, 0) topActivityInfo=ActivityInfo{989420 com.cupoc.omnid.MainActivity} launchCookies=[android.os.BinderProxy@ed901d9] positionInParent=Point(0, 0) parentTaskId=-1 isFocused=false isVisible=false isVisibleRequested=false isSleeping=false locusId=null displayAreaFeatureId=1 isTopActivityTransparent=false appCompatTaskInfo=AppCompatTaskInfo { topActivityInSizeCompat=false topActivityEligibleForLetterboxEducation= false isLetterboxDoubleTapEnabled= false topActivityEligibleForUserAspectRatioButton= false topActivityBoundsLetterboxed= false isFromLetterboxDoubleTap= false topActivityLetterboxVerticalPosition= -1 topActivityLetterboxHorizontalPosition= -1 topActivityLetterboxWidth=1080 topActivityLetterboxHeight=2400 isUserFullscreenOverrideEnabled=false isSystemFullscreenOverrideEnabled=false cameraCompatControlState=hidden}}, pipTask = null, remoteTransition = RemoteTransition { remoteTransition = android.window.IRemoteTransition$Stub$Proxy@7cc179e, appThread = android.app.IApplicationThread$Stub$Proxy@34c627f, debugName = QuickstepLaunch }, displayChange = null, flags = 0, debugId = 15063 }
2024-08-22 20:53:21.699 10663-10663 SearchSessionManager    pid-10663                            D  notifyEvent: query=o at 2860230277 Bundle{timeout=200 entry=2 } targetId=APPLICATION:0:com.cupoc.omnid:com.cupoc.omnid.MainActivity eventType=5
2024-08-22 20:53:21.713  1724-18149 CoreBackPreview         pid-1724                             D  Window{6fd7da1 u0 Splash Screen com.cupoc.omnid}: Setting back callback OnBackInvokedCallbackInfo{mCallback=android.window.IOnBackInvokedCallback$Stub$Proxy@dbc0523, mPriority=0, mIsAnimationCallback=false}
2024-08-22 20:53:21.715   710-710   gralloc4                pid-710                              W  Unable to set buffer name VRI[omnid]#18289(BLAST Consumer)18289: File name too long
2024-08-22 20:53:21.716   710-710   gralloc4                pid-710                              W  Unable to set buffer name VRI[omnid]#18289(BLAST Consumer)18289: File name too long
2024-08-22 20:53:21.717  1724-1790  ActivityManager         pid-1724                             I  Start proc 12057:com.cupoc.omnid/u0a516 for next-top-activity {com.cupoc.omnid/com.cupoc.omnid.MainActivity}
2024-08-22 20:53:21.717   710-710   gralloc4                pid-710                              W  Unable to set buffer name VRI[omnid]#18289(BLAST Consumer)18289: File name too long
2024-08-22 20:53:21.718   710-710   gralloc4                pid-710                              W  Unable to set buffer name VRI[omnid]#18289(BLAST Consumer)18289: File name too long
2024-08-22 20:53:21.720  1724-1766  WindowManager           pid-1724                             V  Sent Transition (#15063) createdAt=08-22 20:53:21.688 via request=TransitionRequestInfo { type = OPEN, triggerTask = TaskInfo{userId=0 taskId=10691 displayId=0 isRunning=true baseIntent=Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.cupoc.omnid/.MainActivity } baseActivity=ComponentInfo{com.cupoc.omnid/com.cupoc.omnid.MainActivity} topActivity=ComponentInfo{com.cupoc.omnid/com.cupoc.omnid.MainActivity} origActivity=null realActivity=ComponentInfo{com.cupoc.omnid/com.cupoc.omnid.MainActivity} numActivities=1 lastActiveTime=2860245345 supportsMultiWindow=true resizeMode=1 isResizeable=true minWidth=-1 minHeight=-1 defaultMinSize=220 token=WCT{RemoteToken{f3fd07f Task{38bea69 #10691 type=standard A=10516:com.cupoc.omnid}}} topActivityType=1 pictureInPictureParams=null shouldDockBigOverlays=false launchIntoPipHostTaskId=-1 lastParentTaskIdBeforePip=-1 displayCutoutSafeInsets=Rect(0, 132 - 0, 0) topActivityInfo=ActivityInfo{6ba84f0 com.cupoc.omnid.MainActivity} launchCookies=[android.os.BinderProxy@6d30095] positionInParent=Point(0, 0) parentTaskId=-1 isFocused=false isVisible=false isVisibleRequested=false isSleeping=false locusId=null displayAreaFeatureId=1 isTopActivityTransparent=false appCompatTaskInfo=AppCompatTaskInfo { topActivityInSizeCompat=false topActivityEligibleForLetterboxEducation= false isLetterboxDoubleTapEnabled= false topActivityEligibleForUserAspectRatioButton= false topActivityBoundsLetterboxed= false isFromLetterboxDoubleTap= false topActivityLetterboxVerticalPosition= -1 topActivityLetterboxHorizontalPosition= -1 topActivityLetterboxWidth=1080 topActivityLetterboxHeight=2400 isUserFullscreenOverrideEnabled=false isSystemFullscreenOverrideEnabled=false cameraCompatControlState=hidden}}, pipTask = null, remoteTransition = RemoteTransition { remoteTransition = android.window.IRemoteTransition$Stub$Proxy@a42d5aa, appThread = android.app.IApplicationThread$Stub$Proxy@70af59b, debugName = QuickstepLaunch }, displayChange = null, flags = 0, debugId = 15063 }
2024-08-22 20:53:21.720  1724-1766  WindowManager           pid-1724                             V      info={id=15063 t=OPEN f=0x0 trk=0 r=[0@Point(0, 0)] c=[
                                                                                                            {WCT{RemoteToken{f3fd07f Task{38bea69 #10691 type=standard A=10516:com.cupoc.omnid}}} m=OPEN f=NONE leash=Surface(name=Task=10691)/@0x7373fd9 sb=Rect(0, 0 - 1080, 2400) eb=Rect(0, 0 - 1080, 2400) d=0},
                                                                                                            {WCT{RemoteToken{7fc124a Task{3e69cc6 #1 type=undefined}}} m=TO_BACK f=SHOW_WALLPAPER leash=Surface(name=Task=1)/@0x83ded21 sb=Rect(0, 0 - 1080, 2400) eb=Rect(0, 0 - 1080, 2400) d=0}
                                                                                                        ]}
2024-08-22 20:53:21.720   710-710   gralloc4                pid-710                              W  Unable to set buffer name VRI[omnid]#18289(BLAST Consumer)18289: File name too long
2024-08-22 20:53:21.724 12057-12057 com.cupoc.omnid         pid-12057                            I  Using CollectorTypeCMC GC.
2024-08-22 20:53:21.728 12057-12057 com.cupoc.omnid         pid-12057                            E  Not starting debugger since process cannot load the jdwp agent.
2024-08-22 20:53:21.740 12057-12057 nativeloader            pid-12057                            D  Configuring clns-6 for other apk /system_ext/framework/com.google.android.camerax.extensions.jar. target_sdk_version=34, uses_libraries=ALL, library_path=/data/app/~~9vzzuPxKNyjSO0uoZjXI2A==/com.cupoc.omnid-KXZ6zW0M5U6g_gw34gdo-A==/lib/arm64:/data/app/~~9vzzuPxKNyjSO0uoZjXI2A==/com.cupoc.omnid-KXZ6zW0M5U6g_gw34gdo-A==/base.apk!/lib/arm64-v8a, permitted_path=/data:/mnt/expand:/data/user/0/com.cupoc.omnid
2024-08-22 20:53:21.741 12057-12057 nativeloader            pid-12057                            D  Configuring clns-7 for other apk /data/app/~~9vzzuPxKNyjSO0uoZjXI2A==/com.cupoc.omnid-KXZ6zW0M5U6g_gw34gdo-A==/base.apk. target_sdk_version=34, uses_libraries=, library_path=/data/app/~~9vzzuPxKNyjSO0uoZjXI2A==/com.cupoc.omnid-KXZ6zW0M5U6g_gw34gdo-A==/lib/arm64:/data/app/~~9vzzuPxKNyjSO0uoZjXI2A==/com.cupoc.omnid-KXZ6zW0M5U6g_gw34gdo-A==/base.apk!/lib/arm64-v8a, permitted_path=/data:/mnt/expand:/data/user/0/com.cupoc.omnid
2024-08-22 20:53:21.745 12057-12057 GraphicsEnvironment     pid-12057                            V  com.cupoc.omnid is not listed in per-application setting
2024-08-22 20:53:21.762 12057-12057 SoLoader                pid-12057                            W  Recording new base apk path: /data/app/~~9vzzuPxKNyjSO0uoZjXI2A==/com.cupoc.omnid-KXZ6zW0M5U6g_gw34gdo-A==/base.apk
                                                                                                    Previously recorded 0 base apk paths.
2024-08-22 20:53:21.773 12057-12086 vulkan                  pid-12057                            D  searching for layers in '/data/app/~~9vzzuPxKNyjSO0uoZjXI2A==/com.cupoc.omnid-KXZ6zW0M5U6g_gw34gdo-A==/lib/arm64'
2024-08-22 20:53:21.773 12057-12086 vulkan                  pid-12057                            D  searching for layers in '/data/app/~~9vzzuPxKNyjSO0uoZjXI2A==/com.cupoc.omnid-KXZ6zW0M5U6g_gw34gdo-A==/base.apk!/lib/arm64-v8a'
2024-08-22 20:53:21.818  1724-27338 CoreBackPreview         pid-1724                             D  Window{ec94567 u0 com.cupoc.omnid/com.cupoc.omnid.MainActivity}: Setting back callback OnBackInvokedCallbackInfo{mCallback=android.window.IOnBackInvokedCallback$Stub$Proxy@1ec1880, mPriority=0, mIsAnimationCallback=false}
2024-08-22 20:53:21.824 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201824,"message":"embeddedAssetFileMap: 778ffc9fe8773a878e9c30a6304784de,png => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/778ffc9fe8773a878e9c30a6304784de.png","code":"None","level":"info"}
2024-08-22 20:53:21.824 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201824,"message":"embeddedAssetFileMap: 376d6a4c7f622917c39feb23671ef71d,png => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/376d6a4c7f622917c39feb23671ef71d.png","code":"None","level":"info"}
2024-08-22 20:53:21.824 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201824,"message":"embeddedAssetFileMap: c79c3606a1cf168006ad3979763c7e0c,png => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/c79c3606a1cf168006ad3979763c7e0c.png","code":"None","level":"info"}
2024-08-22 20:53:21.824 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201824,"message":"embeddedAssetFileMap: 02bc1fa7c0313217bde2d65ccbff40c9,png => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/02bc1fa7c0313217bde2d65ccbff40c9.png","code":"None","level":"info"}
2024-08-22 20:53:21.824 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201824,"message":"embeddedAssetFileMap: 35ba0eaec5a4f5ed12ca16fabeae451d,png => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/35ba0eaec5a4f5ed12ca16fabeae451d.png","code":"None","level":"info"}
2024-08-22 20:53:21.825 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201825,"message":"embeddedAssetFileMap: 5223c8d9b0d08b82a5670fb5f71faf78,png => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/5223c8d9b0d08b82a5670fb5f71faf78.png","code":"None","level":"info"}
2024-08-22 20:53:21.825 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201825,"message":"embeddedAssetFileMap: 563d5e3294b67811d0a1aede6f601e30,png => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/563d5e3294b67811d0a1aede6f601e30.png","code":"None","level":"info"}
2024-08-22 20:53:21.825 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201825,"message":"embeddedAssetFileMap: b6c297a501e289394b0bc5dc69c265e6,png => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/b6c297a501e289394b0bc5dc69c265e6.png","code":"None","level":"info"}
2024-08-22 20:53:21.825 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201825,"message":"embeddedAssetFileMap: 5974eb3e1c5314e8d5a822702d7d0740,png => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/5974eb3e1c5314e8d5a822702d7d0740.png","code":"None","level":"info"}
2024-08-22 20:53:21.825 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201825,"message":"embeddedAssetFileMap: 9d9c5644f55c2f6e4b7f247c378b2fe9,png => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/9d9c5644f55c2f6e4b7f247c378b2fe9.png","code":"None","level":"info"}
2024-08-22 20:53:21.825 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201825,"message":"embeddedAssetFileMap: 47f13fcfaf4f2d8bbb3423adc030c12f,zkey => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/47f13fcfaf4f2d8bbb3423adc030c12f.zkey","code":"None","level":"info"}
2024-08-22 20:53:21.825 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201825,"message":"embeddedAssetFileMap: 486bd5e1dba1debf9273e6742edf9c9f,wasm => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/486bd5e1dba1debf9273e6742edf9c9f.wasm","code":"None","level":"info"}
2024-08-22 20:53:21.848  1724-1766  ActivityTaskManager     pid-1724                             I  Displayed com.cupoc.omnid/.MainActivity for user 0: +161ms
2024-08-22 20:53:21.898 12057-12100 AndroidRuntime          pid-12057                            E  FATAL EXCEPTION: DefaultDispatcher-worker-1
                                                                                                    Process: com.cupoc.omnid, PID: 12057
                                                                                                    java.lang.reflect.InvocationTargetException
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at T6.g.A(SourceFile:43)
                                                                                                    	at T6.g.B(SourceFile:34)
                                                                                                    	at T6.g.t(SourceFile:1)
                                                                                                    	at T6.f.run(SourceFile:1)
                                                                                                    	at expo.modules.updates.UpdatesPackage$c$b.o(SourceFile:13)
                                                                                                    	at I9.a.d(SourceFile:12)
                                                                                                    	at kb.P.run(SourceFile:129)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:959)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:100)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:232)
                                                                                                    	at android.os.Looper.loop(Looper.java:317)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8592)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:878)
                                                                                                    	Suppressed: nb.i: [B0{Cancelling}@bd5db56, Dispatchers.IO]
                                                                                                    Caused by: java.lang.NoSuchFieldException: No field mDefaultJSExceptionHandler in class Lcom/facebook/react/devsupport/u; (declaration of 'com.facebook.react.devsupport.u' appears in /data/app/~~9vzzuPxKNyjSO0uoZjXI2A==/com.cupoc.omnid-KXZ6zW0M5U6g_gw34gdo-A==/base.apk)
                                                                                                    	at java.lang.Class.getDeclaredField(Native Method)
                                                                                                    	at f9.d.n(SourceFile:24)
                                                                                                    	at f9.d.m(SourceFile:9)
                                                                                                    	at f9.d.q(SourceFile:9)
                                                                                                    	at k9.k.E(SourceFile:13)
                                                                                                    	at expo.modules.updates.b.b(SourceFile:8)
                                                                                                    	at expo.modules.updates.UpdatesPackage$d.b(SourceFile:12)
                                                                                                    	at T6.i.createReactInstanceManager(SourceFile:57)
                                                                                                    	at com.facebook.react.M.c(SourceFile:15)
                                                                                                    	at com.facebook.react.x.i(SourceFile:50)
                                                                                                    	at com.facebook.react.t.loadApp(SourceFile:3)
                                                                                                    	... 16 more
2024-08-22 20:53:21.898  1724-2620  ImeTracker              pid-1724                             I  com.cupoc.omnid:94eaffc2: onRequestHide at ORIGIN_SERVER reason HIDE_UNSPECIFIED_WINDOW fromUser false
2024-08-22 20:53:21.902  1724-27338 ActivityTaskManager     pid-1724                             W    Force finishing activity com.cupoc.omnid/.MainActivity
2024-08-22 20:53:21.911 12057-12057 TransactionExecutor     pid-12057                            E  Failed to execute the transaction: tId:1361371967 ClientTransaction{
                                                                                                    tId:1361371967   transactionItems=[
                                                                                                    tId:1361371967     PauseActivityItem{mActivityToken=android.os.BinderProxy@dedfa13,finished=true,userLeaving=false,configChanges=0,dontReport=false,autoEnteringPip=false}
                                                                                                    tId:1361371967     Target activity: com.cupoc.omnid.MainActivity
                                                                                                    tId:1361371967   ]
                                                                                                    tId:1361371967 }
2024-08-22 20:53:21.914 25520-25520 GoogleInpu...hodService pid-25520                            I  GoogleInputMethodService.onStartInput():1982 onStartInput(EditorInfo{EditorInfo{packageName=com.cupoc.omnid, inputType=0, inputTypeString=NULL, enableLearning=false, autoCorrection=false, autoComplete=false, imeOptions=0, privateImeOptions=null, actionName=UNSPECIFIED, actionLabel=null, initialSelStart=-1, initialSelEnd=-1, initialCapsMode=0, label=null, fieldId=0, fieldName=null, extras=null, hintText=null, hintLocales=[]}}, false)
2024-08-22 20:53:21.915  1724-24000 PackageConfigPersister  pid-1724                             W  App-specific configuration not found for packageName: com.cupoc.omnid and userId: 0
2024-08-22 20:53:21.920 10663-10663 ImeTracker              pid-10663                            I  com.cupoc.omnid:94eaffc2: onCancelled at PHASE_CLIENT_ANIMATION_CANCEL
2024-08-22 20:53:21.937  1724-24000 ActivityManager         pid-1724                             I  Process com.cupoc.omnid (pid 12057) has died: prcp TOP 
2024-08-22 20:53:21.937  1724-1745  WindowManager           pid-1724                             I  WIN DEATH: Window{ec94567 u0 com.cupoc.omnid/com.cupoc.omnid.MainActivity}
2024-08-22 20:53:21.937  1724-1745  InputManager-JNI        pid-1724                             W  Input channel object 'ec94567 com.cupoc.omnid/com.cupoc.omnid.MainActivity (client)' was disposed without first being removed with the input manager!
2024-08-22 20:53:21.938  1724-1745  WindowManager           pid-1724                             V  Loading animations: layout params pkg=com.cupoc.omnid resId=0x1030308
2024-08-22 20:53:21.939  1724-18149 ImeTracker              pid-1724                             I  com.cupoc.omnid:49295d04: onRequestHide at ORIGIN_SERVER reason HIDE_REMOVE_CLIENT fromUser false
2024-08-22 20:53:21.939  1724-18149 ImeTracker              pid-1724                             I  com.cupoc.omnid:49295d04: onCancelled at PHASE_SERVER_SHOULD_HIDE
2024-08-22 20:53:21.945  1724-1771  WindowManager           pid-1724                             W  Failed to deliver inset control state change to w=Window{ec94567 u0 com.cupoc.omnid/com.cupoc.omnid.MainActivity EXITING}
                                                                                                    android.os.DeadObjectException
                                                                                                    	at android.os.BinderProxy.transactNative(Native Method)
                                                                                                    	at android.os.BinderProxy.transact(BinderProxy.java:586)
                                                                                                    	at android.view.IWindow$Stub$Proxy.insetsControlChanged(IWindow.java:485)
                                                                                                    	at com.android.server.wm.WindowState.notifyInsetsControlChanged(WindowState.java:3775)
                                                                                                    	at com.android.server.wm.InsetsStateController.lambda$notifyPendingInsetsControlChanged$3(InsetsStateController.java:376)
                                                                                                    	at com.android.server.wm.InsetsStateController.$r8$lambda$MnP7XA1MDcyrUwoj0hcoT5hTO8E(InsetsStateController.java:0)
                                                                                                    	at com.android.server.wm.InsetsStateController$$ExternalSyntheticLambda0.run(R8$$SyntheticClass:0)
                                                                                                    	at com.android.server.wm.WindowAnimator.executeAfterPrepareSurfacesRunnables(WindowAnimator.java:314)
                                                                                                    	at com.android.server.wm.WindowAnimator.animate(WindowAnimator.java:207)
                                                                                                    	at com.android.server.wm.WindowAnimator.lambda$new$1(WindowAnimator.java:99)
                                                                                                    	at com.android.server.wm.WindowAnimator.$r8$lambda$AS_wbK9i-bc6ocCFop7s9PnXP80(WindowAnimator.java:0)
                                                                                                    	at com.android.server.wm.WindowAnimator$$ExternalSyntheticLambda1.doFrame(R8$$SyntheticClass:0)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1404)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1415)
                                                                                                    	at android.view.Choreographer.doCallbacks(Choreographer.java:1015)
                                                                                                    	at android.view.Choreographer.doFrame(Choreographer.java:941)
                                                                                                    	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1389)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:959)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:100)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:232)
                                                                                                    	at android.os.Looper.loop(Looper.java:317)
                                                                                                    	at android.os.HandlerThread.run(HandlerThread.java:85)
                                                                                                    	at com.android.server.ServiceThread.run(ServiceThread.java:46)
2024-08-22 20:53:21.970  1724-1766  WindowManager           pid-1724                             V      info={id=15064 t=CLOSE f=0x10 trk=0 r=[0@Point(0, 0)] c=[
                                                                                                            {WCT{RemoteToken{7fc124a Task{3e69cc6 #1 type=undefined}}} m=TO_FRONT f=SHOW_WALLPAPER|MOVE_TO_TOP leash=Surface(name=Task=1)/@0x83ded21 sb=Rect(0, 0 - 1080, 2400) eb=Rect(0, 0 - 1080, 2400) d=0},
                                                                                                            {WCT{RemoteToken{f3fd07f Task{38bea69 #10691 type=standard A=10516:com.cupoc.omnid}}} m=CLOSE f=NONE leash=Surface(name=Task=10691)/@0x7373fd9 sb=Rect(0, 0 - 1080, 2400) eb=Rect(0, 0 - 1080, 2400) d=0}
                                                                                                        ]}
2024-08-22 20:53:21.984  1724-1771  WindowManager           pid-1724                             W  Failed to deliver inset control state change to w=Window{ec94567 u0 com.cupoc.omnid/com.cupoc.omnid.MainActivity EXITING}
                                                                                                    android.os.DeadObjectException
                                                                                                    	at android.os.BinderProxy.transactNative(Native Method)
                                                                                                    	at android.os.BinderProxy.transact(BinderProxy.java:586)
                                                                                                    	at android.view.IWindow$Stub$Proxy.insetsControlChanged(IWindow.java:485)
                                                                                                    	at com.android.server.wm.WindowState.notifyInsetsControlChanged(WindowState.java:3775)
                                                                                                    	at com.android.server.wm.InsetsStateController.lambda$notifyPendingInsetsControlChanged$3(InsetsStateController.java:376)
                                                                                                    	at com.android.server.wm.InsetsStateController.$r8$lambda$MnP7XA1MDcyrUwoj0hcoT5hTO8E(InsetsStateController.java:0)
                                                                                                    	at com.android.server.wm.InsetsStateController$$ExternalSyntheticLambda0.run(R8$$SyntheticClass:0)
                                                                                                    	at com.android.server.wm.WindowAnimator.executeAfterPrepareSurfacesRunnables(WindowAnimator.java:314)
                                                                                                    	at com.android.server.wm.WindowAnimator.animate(WindowAnimator.java:207)
                                                                                                    	at com.android.server.wm.WindowAnimator.lambda$new$1(WindowAnimator.java:99)
                                                                                                    	at com.android.server.wm.WindowAnimator.$r8$lambda$AS_wbK9i-bc6ocCFop7s9PnXP80(WindowAnimator.java:0)
                                                                                                    	at com.android.server.wm.WindowAnimator$$ExternalSyntheticLambda1.doFrame(R8$$SyntheticClass:0)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1404)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1415)
                                                                                                    	at android.view.Choreographer.doCallbacks(Choreographer.java:1015)
                                                                                                    	at android.view.Choreographer.doFrame(Choreographer.java:941)
                                                                                                    	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1389)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:959)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:100)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:232)
                                                                                                    	at android.os.Looper.loop(Looper.java:317)
                                                                                                    	at android.os.HandlerThread.run(HandlerThread.java:85)
                                                                                                    	at com.android.server.ServiceThread.run(ServiceThread.java:46)
2024-08-22 20:53:21.994 10663-10663 HotseatPre...Controller pid-10663                            D  Predicted items: [WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{com.cupoc.omnid/com.cupoc.omnid.MainActivity} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=Omnid), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{com.google.android.apps.nbu.files/com.google.android.apps.nbu.files.home.HomeActivity} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=Files), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{slowscript.httpfileserver/slowscript.httpfileserver.MainActivity} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=HTTP File Server), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{com.whatsapp/com.whatsapp.Main} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=WhatsApp), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{com.google.android.gm/com.google.android.gm.ConversationListActivityGmail} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=Gmail), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{com.instagram.android/com.instagram.android.activity.MainTabActivity} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=Instagram), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{com.spotify.music/com.spotify.music.MainActivity} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=Spotify), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{com.android.vending/com.android.vending.AssetBrowserActivity} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=Play Store), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{com.google.android.youtube/com.google.android.youtube.app.honeycomb.Shell$HomeActivity} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=YouTube), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{com.grofers.customerapp.lit/com.grofers.customerapp.DEFAULT} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=Blinkit Lit), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{com.trello/com.trello.home.HomeActivity} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=Trello), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{me.rainbow/me.rainbow.MainActivityog} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=Rainbow), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{com.google.android.apps.nbu.paisa.user/com.google.nbu.paisa.flutter.gpay.app.LauncherActivity} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=GPay), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{in.swiggy.android/in.swiggy.android.activities.HomeActivity} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserH
2024-08-22 20:53:22.208  1724-27356 CoreBackPreview         pid-1724                             D  Window{6fd7da1 u0 Splash Screen com.cupoc.omnid EXITING}: Setting back callback null
2024-08-22 20:53:22.209  1724-27338 InputManager-JNI        pid-1724                             W  Input channel object '6fd7da1 Splash Screen com.cupoc.omnid (client)' was disposed without first being removed with the input manager!
2024-08-22 20:53:22.288  1724-27356 WindowManager           pid-1724                             W  Exception thrown during dispatchAppVisibility Window{ec94567 u0 com.cupoc.omnid/com.cupoc.omnid.MainActivity EXITING}
                                                                                                    android.os.DeadObjectException
                                                                                                    	at android.os.BinderProxy.transactNative(Native Method)
                                                                                                    	at android.os.BinderProxy.transact(BinderProxy.java:586)
                                                                                                    	at android.view.IWindow$Stub$Proxy.dispatchAppVisibility(IWindow.java:552)
                                                                                                    	at com.android.server.wm.WindowState.sendAppVisibilityToClients(WindowState.java:3217)
                                                                                                    	at com.android.server.wm.WindowContainer.sendAppVisibilityToClients(WindowContainer.java:1293)
                                                                                                    	at com.android.server.wm.WindowToken.setClientVisible(WindowToken.java:403)
                                                                                                    	at com.android.server.wm.ActivityRecord.setClientVisible(ActivityRecord.java:7100)
                                                                                                    	at com.android.server.wm.ActivityRecord.postApplyAnimation(ActivityRecord.java:5820)
                                                                                                    	at com.android.server.wm.ActivityRecord.commitVisibility(ActivityRecord.java:5762)
                                                                                                    	at com.android.server.wm.Transition.finishTransition(Transition.java:1257)
                                                                                                    	at com.android.server.wm.TransitionController.finishTransition(TransitionController.java:925)
                                                                                                    	at com.android.server.wm.WindowOrganizerController.finishTransition(WindowOrganizerController.java:489)
                                                                                                    	at android.window.IWindowOrganizerController$Stub.onTransact(IWindowOrganizerController.java:278)
                                                                                                    	at com.android.server.wm.WindowOrganizerController.onTransact(WindowOrganizerController.java:199)
                                                                                                    	at android.os.Binder.execTransactInternal(Binder.java:1505)
                                                                                                    	at android.os.Binder.execTransact(Binder.java:1444)

```

--- a/packages/expo-image/CHANGELOG.md
+++ b/packages/expo-image/CHANGELOG.md
@@ -15,6 +15,7 @@
 ### 🐛 Bug fixes
 
 - [Android] Fixed `resolvedLayoutDirection` building issues when using react-native 0.75.X. ([#31062](https://github.com/expo/expo/pull/31062) by [@gabrieldonadel](https://github.com/gabrieldonadel))
+- [Android] Fix `NoSuchMethodException` when running react-native 0.75.X in release mode with Proguard enabled.
 
 ## 1.12.13 — 2024-07-16
 

--- a/packages/expo-image/CHANGELOG.md
+++ b/packages/expo-image/CHANGELOG.md
@@ -15,7 +15,7 @@
 ### 🐛 Bug fixes
 
 - [Android] Fixed `resolvedLayoutDirection` building issues when using react-native 0.75.X. ([#31062](https://github.com/expo/expo/pull/31062) by [@gabrieldonadel](https://github.com/gabrieldonadel))
-- [Android] Fix `NoSuchMethodException` when running react-native 0.75.X in release mode with Proguard enabled.
+- [Android] Fix `NoSuchMethodException` when running react-native 0.75.X in release mode with Proguard enabled. ([#31153](https://github.com/expo/expo/pull/31153) by [@alanjhughes](https://github.com/alanjhughes))
 
 ## 1.12.13 — 2024-07-16
 

--- a/packages/expo-image/android/build.gradle
+++ b/packages/expo-image/android/build.gradle
@@ -15,11 +15,16 @@ buildscript {
 }
 
 android {
+  def rnVersion = getRNVersion()
   namespace "expo.modules.image"
   defaultConfig {
     versionCode 1
     versionName "1.12.14"
-    consumerProguardFiles("proguard-rules.pro")
+    if (rnVersion >= versionToNumber(0, 75, 0)) {
+      consumerProguardFiles('proguard-rules.pro', 'proguard-rules-75.pro')
+    } else {
+      consumerProguardFiles('proguard-rules.pro')
+    }
 
     buildConfigField("boolean", "ALLOW_GLIDE_LOGS", project.properties.get("EXPO_ALLOW_GLIDE_LOGS", "false"))
   }
@@ -53,3 +58,29 @@ dependencies {
   implementation 'org.jetbrains.kotlinx:kotlinx-coroutines-core:1.5.1'
   implementation "jp.wasabeef:glide-transformations:4.3.0"
 }
+
+def versionToNumber(major, minor, patch) {
+  return patch * 100 + minor * 10000 + major * 1000000
+}
+
+def getNodeModulesPackageVersion(packageName, overridePropName) {
+  def nodeModulesVersion = providers.exec {
+    workingDir(projectDir)
+    commandLine("node", "-e", "console.log(require('$packageName/package.json').version);")
+  }.standardOutput.asText.get().trim()
+  def version = safeExtGet(overridePropName, nodeModulesVersion)
+
+  def coreVersion = version.split("-")[0]
+  def (major, minor, patch) = coreVersion.tokenize('.').collect { it.toInteger() }
+
+  return versionToNumber(
+      major,
+      minor,
+      patch
+  )
+}
+
+def getRNVersion() {
+  return getNodeModulesPackageVersion("react-native", "reactNativeVersion")
+}
+

--- a/packages/expo-image/android/build.gradle
+++ b/packages/expo-image/android/build.gradle
@@ -83,4 +83,3 @@ def getNodeModulesPackageVersion(packageName, overridePropName) {
 def getRNVersion() {
   return getNodeModulesPackageVersion("react-native", "reactNativeVersion")
 }
-

--- a/packages/expo-image/android/proguard-rules-75.pro
+++ b/packages/expo-image/android/proguard-rules-75.pro
@@ -1,0 +1,3 @@
+-keepclassmembers class com.facebook.react.uimanager.drawable.CSSBackgroundDrawable {
+   public void setLayoutDirectionOverride(int);
+}

--- a/packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageView.kt
+++ b/packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageView.kt
@@ -219,9 +219,12 @@ class ExpoImageView(
 
       borderDrawable.apply {
         val setLayoutDirectionMethod = try {
+          // 0.74
           ReactViewBackgroundDrawable::class.java.getDeclaredMethod("setResolvedLayoutDirection", Int::class.java)
         } catch (e: NoSuchMethodException) {
-          ReactViewBackgroundDrawable::class.java.getMethod("setLayoutDirectionOverride", Int::class.java)
+          // 0.75
+          val clazz = Class.forName("com.facebook.react.uimanager.drawable.CSSBackgroundDrawable")
+          clazz.getDeclaredMethod("setLayoutDirectionOverride", Int::class.java)
         }
         setLayoutDirectionMethod.invoke(this, newLayoutDirection)
         setBounds(0, 0, width, height)

